python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint_search.py

```diff
@@ -906,89 +1042,145 @@ def _boresch_restraint_BSS(
 
-        # calculate average and SD
-        for pair in anchors_dict.keys():
-            anchors_dict[pair]["avg_dist"] = anchors_dict[pair]["dists"].mean()
-            anchors_dict[pair]["sd_dist"] = anchors_dict[pair]["dists"].std()
+            lig_selection_str: str
+                The selection string for the atoms in the ligand to consider
+                as potential anchor points.
 
-        # get n pairs with lowest SD
-        pairs_ordered_sd = []
-        for item in sorted(anchors_dict.items(), key=lambda item: item[1]["sd_dist"]):
-            pairs_ordered_sd.append(item[0])
-            # print(f'Pair: {item[0]}, av distance: {item[1]["avg_dist"]:.2f}, SD: {item[1]["sd_dist"]:.2f}')
+            recept_selection_str: str
+                The selection string for the protein in the ligand to consider
+                as potential anchor points.
 
-        # Print out pairs with lowest SD
-        # print("The ligand-protein atom pairs with the lowest SD in distance are:")
-        # for i in range(5):
-        #    print(f"{u.atoms[pairs_ordered_sd[i][0]]} and {u.atoms[pairs_ordered_sd[i][1]]}")
+            cutoff: BioSimSpace.Types.Length
+                The greatest distance between ligand and receptor anchor atoms.
+                Only affects behaviour when method == "BSS" Receptor anchors
+                further than cutoff Angstroms from the closest ligand anchors will not
+                be included in the search for potential anchor points.
 
-        ### For Pairs with Lowest Pairwise RMSDs, find Adjacent Heavy Atoms
+            Returns
+            -------
 
-        def getAnchorAts(a1_idx, u):
+            pairs_ordered_sd : list of tuples
+                List of receptor-ligand atom pairs ordered by increasing variance of distance over
+                the trajectory.
             """
-            Takes in index of anchor atom 1 and universe and returns
-            list of all three anchor atoms, which are chosen to be bonded
-            and not H".
 
-            Args:
-                a1_idx (int): Index of the first anchor atom
-                u (mda universe): The mda universe
+            lig_selection = u.select_atoms(lig_selection_str)
+            pair_variance_dict = {}
+
+            # Get all receptor atoms within specified distance of cutoff
+            for lig_atom in lig_selection:
+                for prot_atom in u.select_atoms(
+                    f"{recept_selection_str} and (around {cutoff / _angstrom} index {lig_atom.index})"
+                ):
+                    pair_variance_dict[(lig_atom.index, prot_atom.index)] = {}
+                    pair_variance_dict[(lig_atom.index, prot_atom.index)]["dists"] = []
+
+            # Compute Average Distance and SD
+            for frame in _tqdm(
+                u.trajectory, desc="Searching for low variance pairs. Frame no: "
+            ):
+                for lig_atom_index, prot_atom_index in pair_variance_dict.keys():
```

Comment:
Why not use https://docs.mdanalysis.org/1.1.1/documentation_pages/lib/distances.html#MDAnalysis.lib.distances.capped_distance

Reply:
Thanks for pointing this out - this looks good. However, since we're aiming to replace the MDAnalysis functionality with the new sire search functionality soon anyway, I'm tempted to leave this.